CMP1903_A1_2324/Testing.cs

```diff
@@ -15,6 +17,59 @@ internal class Testing
          * Use debug.assert() to make the comparisons and tests.
          */
 
-        //Method
+        /// <summary>
+        /// Creates a <c>Die</c> object and runs <see cref="Die.Roll()"/> 100,000 to test if values are valid. 
+        /// Uses Debug.Assert() to check dice value is between 1-6 and throws error if value is outside range.
+        /// </summary>
+        /// 
+        public void TestDie()
+        {
+            // Create a new Die object to test
+            Die die = new Die();
+            
+            // Loop 100,000 times to check if Die object only produces valid values
+            // 100,000 loops should be enough to make the chance of a false-positive practically 0
+            for (int i = 0; i < 100000; i++)
+            {
+                // Rolls the Die
+                die.Roll();
+                // Uses Debug.Assert() to confirm DieValue remains between 1-6
+                Debug.Assert(1 <= die.DieValue && die.DieValue <= 6, "Die value outside of valid range");
```

Comment:
As I mentioned in your review of my code, whilst there is nothing wrong with testing your condition this way; splitting it into two separate tests/asserts would allow you to exactly locate what part of the roll went wrong.